src/backend/api/reservations.js

```diff
@@ -0,0 +1,96 @@
+const express = require("express");
+const router = express.Router();
+const knex = require("../database");
+
+// GET all reservations
+router.get("/", async (req, res) => {
+  try {
+    const reservations = await knex.select().from("reservations");
+    res.json(reservations);
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// GET a reservation by id
+router.get("/:id", async (req, res) => {
+  try {
+    const reservation = await knex
+      .select()
+      .from("reservations")
+      .where("id", req.params.id)
+      .first();
+    if (!reservation) {
+      res.status(404).json({ message: "Reservation not found" });
+      return;
+    }
+    res.json(reservation);
+  } catch (err) {
+    res.status(500).json({ message: err.message });
+  }
+});
+
+// CREATE a new reservation
+router.post("/", async (req, res) => {
+  try {
+    const { name, email, meal_id, created_date } = req.body;
+    if (!name || !email || !meal_id || !created_date) {
+      res
+        .status(400)
+        .json({
+          message: "Name, email, meal_id, and created_date are required",
+        });
+      return;
+    }
```

Comment:
Do you need to check this way or can you rely on database schema and handle the error while inserting?